sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_download.py

```diff
@@ -0,0 +1,53 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+from ._models import FileProperties
+
+
+class StorageStreamDownloader(object):
```

Comment:
Should we move this into the _shared folder?

Reply:
No, as it's a custom ADLS-specific implementation that's dependent on the Blobs-specific implementation.
Though potentially we can refactor a common base to add to the _shared directory later.